src/hooks/card/useCardNumber.ts

```diff
@@ -0,0 +1,25 @@
+import { ChangeEvent, useState } from 'react';
+
+import { isNumber } from '@/utils';
+
+export default function useCardNumber() {
+  const [cardNumber, setCardNumber] = useState({
+    num1: '',
+    num2: '',
+    num3: '',
+    num4: '',
+  });
+
+  const handleChangeCardNumber = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+
+    if (!isNumber(value) && value !== '') return;
+
```

Comment:
```suggestion
    if (!isNumber(value) && value !== '') return;
    if (!(name in cardNumber))
      throw new Error(`input element에 지정되지 않은 name이 주입되었습니다. [injected name with '${name}]'`);
```
(마이너) 이번 미션에서 해당 훅이 크게 확장될 여지가 없어 유효한 코멘트는 아닐수도 있지만 도움이 되실수도 있을 것 같아서 남기는 코멘트에요
요구사항이 자주 변경될여지가 있으면 가드로직을 넣어둬도 좋아요 👀 
hook이 component의 name에 의존하고있다보니 실수할 여지가 조금 있을 것 같기도해서.. 자주변경된다면 이렇게 가드로직을 넣어둬도 좋아요.

Reply:
마이너한 리뷰들도 너무 좋습니다!

hook이 component에 의존성이 있다보니, 기존 코드라면 컴포넌트의 name attribute에는 어떤 값이라도 넣을 수 있으니 개발자가 의도하지 않은 name이 들어오는 경우에도 그대로 state에 값이 반영되었겠네요. 실제로 개발자의 실수로 문제가 생겼을 때, 가드로직이 존재한다면 디버깅을 바로 할 수 있을 것 같습니다!